File: src/main/java/org/ffpy/plugin/coding/action/menu/CaseFormatAction.java
package org.ffpy.plugin.coding.action.menu;

import com.intellij.openapi.actionSystem.AnActionEvent;
import org.ffpy.plugin.coding.action.BaseReplaceAction;
import org.ffpy.plugin.coding.util.MyStringUtils;

/**
 * 驼峰-下划线互转
 */
public class CaseFormatAction extends BaseReplaceAction {

    @Override
    protected String replace(AnActionEvent e, String text) throws Exception {
        return text.contains("_") ? MyStringUtils.underScoreCase2CamelCase(text) :
                MyStringUtils.camelCase2UnderScoreCase(text);
    }
}

File: src/main/kotlin/org/ffpy/plugin/coding/action/menu/CaseFormatAction.kt
package org.ffpy.plugin.coding.action.menu

import com.intellij.openapi.actionSystem.AnActionEvent
import org.ffpy.plugin.coding.action.BaseReplaceAction
import org.ffpy.plugin.coding.util.MyStringUtils.camelCase2UnderScoreCase
import org.ffpy.plugin.coding.util.MyStringUtils.underScoreCase2CamelCase

/**
 * 驼峰-下划线互转
 */
class CaseFormatAction: BaseReplaceAction() {

    override fun replace(e: AnActionEvent, text: String): String? {
        return if (text.contains("_")) underScoreCase2CamelCase(text) else camelCase2UnderScoreCase(text)
    }
}